### PR TITLE
General error message handling for invalid certificates (DEV)

### PR DESCRIPTION
**Pull request by chiljamgossow**
move general certificate error handling to InvalidHealthCertificateException

### CODE REVIEW

**Review by NataliaLemmerth — DISMISSED** [review]
UA approved

**Review by harambasicluka — CHANGES_REQUESTED** [review]
Please change the id only in german and add it to `values` (do not delete the old ones), everything else will be handled by the translation delivery.

File: Corona-Warn-App/src/main/res/values-de/vaccination_strings.xml [hunk 1]
```diff
@@ -109,7 +111,12 @@
     <!-- XBUT: Text for invalid certificate error button, linking to FAQ-->
     <string name="error_button_vc_faq">FAQ zu Zertifikaten</string>
     <!-- XTXT: Explains user about vaccination certificate: URL, has to be "translated" into english (relevant for all languages except german) - https://www.coronawarn.app/en/faq/#vac_cert_invalid -->
-    <string name="error_button_vc_faq_link">https://www.coronawarn.app/de/faq/#hc_prefix_invalid</string>
+    <string name="error_button_vc_faq_link">https://www.coronawarn.app/en/faq/#vac_cert_invalid</string>
+
+    <!-- XBUT: Text for invalid certificate error button, linking to FAQ-->
+    <string name="error_button_dcc_faq">FAQ zu Zertifikaten</string>
+    <!-- XTXT: Explains user about vaccination certificate: URL, has to be "translated" into english (relevant for all languages except german) - https://www.coronawarn.app/en/faq/#vac_cert_invalid -->
+    <string name="error_button_dcc_faq_link">https://www.coronawarn.app/de/faq/#hc_prefix_invalid</string>
```
Comment (mtwalli) [inline Corona-Warn-App/src/main/res/values-de/vaccination_strings.xml:119]: Please add the new strings in `covid_certificates_strings.xml` ?

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/common/exception/InvalidVaccinationCertificateException.kt [hunk 2]
```diff
@@ -57,6 +34,7 @@ class InvalidVaccinationCertificateException(
         }
 }
 
-private const val ERROR_MESSAGE_VC_INVALID = R.string.error_vc_invalid
+private const val PREFIX = "VC_"
```
Comment (mtwalli) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/common/exception/InvalidVaccinationCertificateException.kt:37]: is it for Vaccination certificates error codes or for all types of certificates? and is it required to add this prefix ?
I found very difficult to find the actual error code , if you look for `VC_PREFIX_INVALID` as the error states

Comment (chiljamgossow) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/common/exception/InvalidVaccinationCertificateException.kt:37]: The prefix is part of the tech spec for vaccination - it is only affecting error messages specific to vaccination. The idea is that InvalidVaccinationCertificateException handles specific errors/ specific error messages for vaccination, while InvalidHealthCertificateException has general errors for all certificate types.

**Review by harambasicluka — COMMENTED** [review]
strings look fine! but somehow the linter complains O.o

**Review by mtwalli — DISMISSED** [review]
LGTM

**Review by harambasicluka — APPROVED** [review]

**Pull request merged**